liquibase-core/src/test/groovy/liquibase/datatype/DataTypeFactoryTest.groovy

```diff
@@ -210,8 +210,50 @@ class DataTypeFactoryTest extends Specification {
+        "timestamp"                                    | new PostgresDatabase() | "TIMESTAMP WITHOUT TIME ZONE"                  | TimestampType | false
+        "timestamp(6)"                                 | new PostgresDatabase() | "TIMESTAMP(6) WITHOUT TIME ZONE"               | TimestampType | false
+        "timestamp with timezone"                      | new PostgresDatabase() | "TIMESTAMP WITH TIME ZONE"                     | TimestampType | false
+        "timestamp(6) with timezone"                   | new PostgresDatabase() | "TIMESTAMP(6) WITH TIME ZONE"                  | TimestampType | false
+        "timestamp without timezone"                   | new PostgresDatabase() | "TIMESTAMP WITHOUT TIME ZONE"                  | TimestampType | false
+        "timestamp(6) without timezone"                | new PostgresDatabase() | "TIMESTAMP(6) WITHOUT TIME ZONE"               | TimestampType | false
+        "timestamptz"                                  | new PostgresDatabase() | "TIMESTAMP WITH TIME ZONE"                     | TimestampType | false
+        "timestamptz(6)"                               | new PostgresDatabase() | "TIMESTAMP(6) WITH TIME ZONE"                  | TimestampType | false
+        "java.sql.Timestamp"                           | new PostgresDatabase() | "TIMESTAMP WITHOUT TIME ZONE"                  | TimestampType | false
+        "java.sql.Timestamp(6)"                        | new PostgresDatabase() | "TIMESTAMP(6) WITHOUT TIME ZONE"               | TimestampType | false
+        "java.sql.Types.TIMESTAMP"                     | new PostgresDatabase() | "TIMESTAMP WITHOUT TIME ZONE"                  | TimestampType | false
+        "java.sql.Types.TIMESTAMP(6)"                  | new PostgresDatabase() | "TIMESTAMP(6) WITHOUT TIME ZONE"               | TimestampType | false
+        "java.sql.Types.TIMESTAMP_WITH_TIMEZONE"       | new PostgresDatabase() | "TIMESTAMP WITH TIME ZONE"                     | TimestampType | false
+        "java.sql.Types.TIMESTAMP_WITH_TIMEZONE(6)"    | new PostgresDatabase() | "TIMESTAMP(6) WITH TIME ZONE"                  | TimestampType | false
         "BINARY(16)"                                   | new H2Database()       | "BINARY(16)"                                   | BlobType      | false
+        "timestamp"                                    | new H2Database()       | "TIMESTAMP"                                    | TimestampType | false
+        "timestamp(6)"                                 | new H2Database()       | "TIMESTAMP(6)"                                 | TimestampType | false
+        "TIMESTAMP WITH TIMEZONE"                      | new H2Database()       | "TIMESTAMP WITH TIMEZONE"                      | TimestampType | false
+        "TIMESTAMP(6) WITH TIMEZONE"                   | new H2Database()       | "TIMESTAMP(6) WITH TIMEZONE"                   | TimestampType | false
```

Comment:
H2 cannot parse this due to missing space between `TIME` and `ZONE`.

Reply:
See reply for next comment.